src/khepri_machine.erl

```diff
@@ -1268,39 +1281,41 @@ apply(
     end;
 apply(
   Meta,
-  #unregister_projection{name = ProjectionName},
+  #unregister_projections{names = Names},
   State) ->
+    RemoveProjection = case Names of
+                           all ->
+                               fun(_Name) -> true end;
+                           _ when is_list(Names) ->
+                               Names1 = sets:from_list(Names, [{version, 2}]),
+                               fun(Name) -> sets:is_element(Name, Names1) end
+                       end,
     ProjectionTree = get_projections(State),
-    ProjectionTree1 =
-    khepri_pattern_tree:filtermap(
-      ProjectionTree,
-      fun (_PathPattern, Projections) ->
-              Projections1 =
-              lists:filter(
-                fun (#khepri_projection{name = Name} = Projection)
-                      when Name =:= ProjectionName ->
-                        khepri_projection:delete(Projection),
-                        false;
-                    (_OtherProjection) ->
-                        true
+    {ProjectionTree1, RemovedProjectionsMap} =
+    khepri_pattern_tree:map_fold(
+      fun(Pattern, Projections, Acc) ->
+              {RemovedProjections, Projections1} =
+              lists:partition(
+                fun(Projection) ->
+                        Name = khepri_projection:name(Projection),
+                        RemoveProjection(Name)
                 end, Projections),
-              case Projections1 of
-                  [] ->
-                      false;
-                  _ ->
-                      {true, Projections1}
-              end
-      end),
-    Reply = case ProjectionTree1 of
-                ProjectionTree ->
-                    Info = #{name => ProjectionName},
-                    Reason = ?khepri_error(projection_not_found, Info),
-                    {error, Reason};
-                _ ->
-                    clear_compiled_projection_tree(),
```

Comment:
Don’t we need this call to `clear_compiled_projection_tree()`?

Reply:
Whoops, good catch!